be/src/util/thrift_rpc_helper.cpp

```diff
@@ -59,6 +59,57 @@ void ThriftRpcHelper::setup(ExecEnv* exec_env) {
     _s_exec_env = exec_env;
 }
 
+template <>
+Status ThriftRpcHelper::rpc_impl(std::function<void(ClientConnection<FrontendServiceClient>&)> callback,
+                                 ClientConnection<FrontendServiceClient>& client,
+                                 const TNetworkAddress& address) noexcept {
+    std::stringstream ss;
+    try {
+        callback(client);
+        return Status::OK();
+    } catch (apache::thrift::protocol::TProtocolException& e) {
+        if (e.getType() == apache::thrift::protocol::TProtocolException::TProtocolExceptionType::INVALID_DATA) {
```

Comment:
The new error message still does not allow users to easily understand the source of the error and its possible impact. We can consider let the caller to pass some information for error message. 
For example, when the commit transaction call fails, we need to tell the user the stage of failure, and the result of the current ingestion transaction will be unknown and need to be checked manually.